eclipse-theia/theia
# fix: file-saving dialog with dirty editors

#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/12859

This PR fixes a bug where cancelling the file-saving dialog when closing a dirty editor would actually close the editor.
With the new changes, cancelling the dialog will leave the dirty editor opened, which is the intended behavior.

#### How to test

1. Open Theia application.
2. Create a new untitled text file (`File` -> `New Text File` or `Alt+N`).
3. Try to close the editor by pressing the `X` or `Alt+W`.
4. The file saving dialog should appear -> Press `Cancel`.
5. The editor should stay opened.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
